python_modules/dagster/dagster/_seven/compat/pendulum.py

```diff
@@ -45,7 +45,12 @@ def mock_pendulum_timezone(override_timezone):
 
 def create_pendulum_time(year, month, day, *args, **kwargs):
     if "tz" in kwargs and "dst_rule" in kwargs and _IS_PENDULUM_1:
-        tz = pendulum.timezone(kwargs.pop("tz"))
+        tz_name_or_info = kwargs.pop("tz")
+        tz = (
+            pendulum.timezone(tz_name_or_info)
+            if isinstance(tz_name_or_info, str)
+            else tz_name_or_info
+        )
```

Comment:
is this still needed? what's your confidence level that pendulum plays nicely when passed in a non-pendulum timezone?

Reply:
Pendulum ignores non-pendulum timezones. Basically timezone apis in python have two camps:
- pytz and pendulum
- dateutil.tz and zoneinfo

Pendulum obviously supports the former, but not the latter so it looks to see if the timezone is something it supports.

Reply:
Also I'm 100% confident that might callsite is the only one passing in an actual tzinfo instead of a string because it raised before my changes here.